opal/mca/btl/base/btl_base_select.c

```diff
@@ -107,6 +122,16 @@ int mca_btl_base_select(bool enable_progress_threads,
                                 "select: no init function; ignoring component %s",
                                 component->btl_version.mca_component_name);
         } else {
+#if MEMORY_LINUX_MALLOC_ALIGN_ENABLED
+            if (strcmp(component->btl_version.mca_component_name, "openib") == 0) {
```

Comment:
Ah -- I see the problem: I think I wasn't clear in my comment to you before about doing this "through the framework base."  Sorry.  It was clear in my head! :wink: Here's what I meant:

I was actually referring to the `opal/mca/memory` base (not the BTL base).  I.e., make an OPAL memory framework function to set a memalign hook.  This function would be located in `opal/mca/memory/base`, and openib (and oshmem, and... whoever else needs it) can just call it directly, without needing to directly include `opal/mca/memory/linux/something.h` or directly invoke a function in `opal/mca/memory/linux`.  Instead, the memory/base function will lookup and see if the selected memory component supports setting a memalign hook, and if so, set it (e.g., perhaps by calling a new member function on the memory module structure).  The linux memory hook component will support this functionality; I don't know if the solaris one will (i.e., it needs to be safe for the solaris memory component to *not* support this functionality).

Does that make more sense?